march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator.py

```diff
@@ -50,6 +51,15 @@ def __init__(self, context):
         self.robot = urdf.Robot.from_parameter_server()
         self.gait = GaitFactory.empty_gait(self, self.robot, self.DEFAULT_GAIT_DURATION)
 
+        self.build_ui(context)
+
+        # Initialize the publisher on startup
+        self.set_topic_name(self.topic_name_line_edit.text())
+
+        self.load_gait_into_ui()
+
+    # Called by __init__
+    def build_ui(self, context):
         # Start UI construction
```

Comment:
Yeah I was hesitating whether to leave these comments in. I put them in to make the flow of the program more clear, if you collapse all methods you now clearly see when the program does what (first few methods are launched by __init__, bottom methods are mostly launched by interaction with the gui). If I follow the docstring conventions, the comments should A) be inside the method, which would mean that they cannot be viewed when all methods are collapsed, and B) explain the function of the method, rather than when it is called. 

This resulted in a dilemma on whether I should remove them (since I did not want to create inconsistent code documentation) or not, so I decided to leave them in to see if you had any opinions.

Reply:
Okay I see now that these comments are not describing the actual method. It's fine :+1: